app/workers/publishing_api_document_republishing_worker.rb

```diff
@@ -27,6 +27,27 @@ def perform(document_id, bulk_publishing = false)
     end
   end
 
+  def preview_refreshed_editions(document_id)
+    @document = Document.find(document_id)
+
+    return [] unless document_has_republishable_editions?
+
+    actions.map { |action|
+      return nil if action == :patch_links
+
+      edition = {
+        refresh_latest_unpublished_edition: latest_unpublished_edition,
+        refresh_pre_publication_edition: pre_publication_edition,
+        refresh_published_edition: published_edition,
+        refresh_withdrawn_edition: withdrawn_edition,
+      }[action]
+
+      raise Error if edition.blank? # unknown action
```

Comment:
Is there a way we can rewrite this so that we don't rely on errors being raised to tell us we're missing something? Would some kind of case statement guarantee that all of a defined set of actions are covered? Probably not in typeless Ruby land, right?